src/GitHubApps/src/DarcBot/AzureFunction.cs

```diff
@@ -0,0 +1,212 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Kusto.Ingest;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Extensions.Http;
+using Microsoft.DotNet.Kusto;
+using Microsoft.Extensions.Logging;
+using Octokit;
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+
+namespace DarcBot
+{
+    public static class GitHubWebHook
+    {
+        private static ILogger _log;
+        private static Regex _darcBotIssueIdentifierRegex = new Regex(@"\[BuildId=(?<buildid>[^,]+),RecordId=(?<recordid>[^,]+),Index=(?<index>[^\]]+)\]", RegexOptions.Multiline | RegexOptions.IgnoreCase | RegexOptions.IgnorePatternWhitespace);
+        private static Regex _darcBotPropertyRegex = new Regex(@"\[(?<key>.+)=(?<value>[^\]]+)\]", RegexOptions.Multiline | RegexOptions.IgnoreCase | RegexOptions.IgnorePatternWhitespace);
+        private static readonly string docLink = "[DarcBot documentation](https://github.com/dotnet/arcade-services/tree/master/src/GitHubApps/src/DarcBot/Readme.md)";
+
+        [FunctionName("GitHubWebHook")]
+        public static async Task<IActionResult> Run(
+            [HttpTrigger(AuthorizationLevel.Anonymous, "POST", Route = null)]
+            HttpRequestMessage req, ILogger log)
+        {
+            _log = log;
+
+            dynamic data = await req.Content.ReadAsAsync<object>();
+
+            string userType = data?.comment?.user?.type;
+            if (userType == "Bot")
+            {
+                _log.LogInformation("Comment is from DarcBot, ignoring.");
+                return new OkObjectResult($"Ignoring DarcBot comment");
+            }
+            string gitHubAction = data?.action;
+
+            int installationId = data?.installation?.id;
+            long repositoryId = data?.repository?.id;
+            int prNumber = data?.issue?.number;
+
+            // Check for darcbot directed comments
+            if (gitHubAction == "created")
+            {
+                /* Nothing to do, this is the section where you would handle user comments if they contained explicit directives to darcbot.
```

Comment:
Should we return here? Because this is going to fall into the next if statement. "create" is not "opened", "reopened", or "closed".

Reply:
It really depends on what you want to do here.  I had some logic in here at one point but removed it to simplify the bot.  I feel like the section is useful for "something" in the future and hence I left the block as a user-friendliness thing, but I'm ok with deleting it entirely if it's bothersome.

Reply:
I just wasn't clear that it was intending to fall into the next if.  It _feels_ like an early return or an "else" is missing.  Lots of other people will probably have the same gut feeling.  So something in the comment explaining the fall through behavior might be good.

Or maybe have a `switch(gitHubAction)` and call appropriate helper methods for the various cases would make it more explicit too.